### PR TITLE
Add support of f32 for Gauge and Counter

### PR DESCRIPTION
**Pull request by navaati**

### CODE REVIEW

**Comment by navaati** [comment]
Still gotta edit the changelog.

**Comment by navaati** [comment]
There ya go, changelog done. I took the liberty to merge the entry with @brocaar's, since this PR kinda completes his.

@mxinden I’m a bit afraid about this PR, since it touches Atomic code, but well, I copied the code and searched/replaced. Please take a look !

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,15 +6,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
-- Add `EncodeGaugeValue` `i32` and `f32`, `EncodeCounterValue` `u32` and `f32` and `EncodeExemplarValue` `f32` and `u32` implementations.
-  See [PR 173].
+### Added
+
+- Support `i32`/`f32` for `Gauge` and `u32`/`f32` for `Counter`/`CounterWithExemplar`.
+  See [PR 173] and [PR 216].
 
 [PR 173]: https://github.com/prometheus/client_rust/pull/173
+[PR 216]: https://github.com/prometheus/client_rust/pull/216
```
Comment (mxinden) [inline CHANGELOG.md:25]: Thanks!

**Review by mxinden — APPROVED** [review]
Good work. Atomics code looks fine.

Shall we hold off with a release for now?

File: src/encoding/text.rs [hunk 2]
```diff
@@ -717,9 +717,12 @@ mod tests {
         let mut registry = Registry::default();
         registry.register("my_counter", "My counter", counter);
 
+        let counter_f32 = Counter::<f32, AtomicU32>::default();
+        registry.register("f32_counter", "Counter::<f32, AtomicU32>", counter_f32);
+      
         let counter_u32 = Counter::<u32, AtomicU32>::default();
         registry.register("u32_counter", "Counter::<u32, AtomicU32>", counter_u32);
-
+
```
Comment (mxinden) [inline src/encoding/text.rs:725]:
```suggestion

```

File: src/encoding/text.rs [hunk 3]
```diff
@@ -717,9 +717,12 @@ mod tests {
         let mut registry = Registry::default();
         registry.register("my_counter", "My counter", counter);
 
+        let counter_f32 = Counter::<f32, AtomicU32>::default();
+        registry.register("f32_counter", "Counter::<f32, AtomicU32>", counter_f32);
+
```
Comment (mxinden) [inline src/encoding/text.rs:722]:
```suggestion

```

**Comment by navaati** [comment]
> Shall we hold off with a release for now?

If you have enough content for a release, go for it: the sooner you get it in the wild, the sooner you’ll have feedback if there is a problem (not that I think there would)… It’s a breaking release anyway so nobody will get broken on a `cargo update`.

As for myself I’m happy when I can remove this line https://gitlab.com/navaati/licorne/-/blob/f26f4141c357b41aae2f1235f8fe0cb3bc7af638/firmware/Cargo.toml#L42 :D

**Pull request merged**